include/mesh.h

```diff
@@ -296,10 +309,14 @@ class Mesh {
   Map<Mesh<Tdim>> neighbour_meshes_;
   //! Container of particles
   Container<ParticleBase<Tdim>> particles_;
+  //! Container of particle sets
+  spp::sparse_hash_map<unsigned, Container<ParticleBase<Tdim>>> particle_sets_;
   //! Map of particles for fast retrieval
   Map<ParticleBase<Tdim>> map_particles_;
   //! Container of nodes
   Container<NodeBase<Tdim>> nodes_;
+  //! Container of node sets
+  spp::sparse_hash_map<unsigned, Container<ParticleBase<Tdim>>> node_sets_;
```

Comment:
```suggestion
  spp::sparse_hash_map<unsigned, Container<NodeBase<Tdim>>> node_sets_;
```

Reply:
Hey sorry to go back and forth, after some analysis it might be better to stick to `tsl::robin_map`